### PR TITLE
Minor Optimizations

### PR DESCRIPTION
**Pull request by romulogcerqueira**
In SonarWidget:
- added binary search method to looking for beam index;
- corrected ternary operator syntax;
- small optimizations

### CODE REVIEW

File: src/sonar_widget/SonarPlot.cc [hunk 1]
```diff
@@ -31,7 +31,7 @@ void SonarPlot::setData(const base::samples::Sonar& sonar)
     if (!sonar.beam_count || !sonar.bin_count)
         return;
 
-    sonar.beam_count > 1 ? isMultibeamSonar = true : isMultibeamSonar = false;
+    isMultibeamSonar = sonar.beam_count > 1 ? true : false;
```
Comment (doudou) [inline src/sonar_widget/SonarPlot.cc]: `sonar.beam_count > 1` is already a boolean expression.

`isMultiBeamSonar = (sonar.beam_count > 1)`

File: src/sonar_widget/SonarPlot.cc [hunk 2]
```diff
@@ -276,20 +274,16 @@ void SonarPlot::generateMultibeamTransferTable(const base::samples::Sonar& sonar
...
 
             // pixels in the sonar image
             else {
-                for (unsigned int k = 0; k < sonar.beam_count - 1; k++) {
-                    if (angle >= sonar.bearings[k].rad && angle < sonar.bearings[k + 1].rad) {
-                        transfer.push_back(k * sonar.bin_count + radius);
-                        break;
-                    }
-                }
+                int idBeam = binarySearch(sonar.bearings, 0, sonar.bearings.size() - 1, angle);
```
Comment (doudou) [inline src/sonar_widget/SonarPlot.cc]: If you can do a binary search, it means you assume that the sonar bearings are sorted in increasing order (a fair assumption !). Which also means that the idBeam of a pixel on the line is either the same than the pixel before it, or the next one.

A.k.a.: you don't need to do search *at all*, and definitely have no need for binarySearch.

File: src/sonar_widget/SonarPlot.cc (hunk 1 above)
Comment (romulogcerqueira) [inline src/sonar_widget/SonarPlot.cc]: Ok. Done!

File: src/sonar_widget/SonarPlot.cc (hunk 2 above)
Comment (romulogcerqueira) [inline src/sonar_widget/SonarPlot.cc]: Right. I followed your thought and the binary search is useless for this case. I updated the code.

**Pull request merged**